src/services/tyeProcessProvider.ts

```diff
@@ -60,26 +79,48 @@ export default class LocalTyeProcessProvider implements TyeProcessProvider {
         return this.processes.pipe(first()).toPromise();
     }
 
-    private async getProcessList(): Promise<TyeProcess[]> {
+    private async getTyeProcessesWithPorts(): Promise<TyeProcessWithPorts[]> {
         const tyePath = await this.tyePathProvider.getTyePath();
         const tyeProcesses = await this.processProvider.listProcesses(tyePath);
-        const tyeProcessesWithPorts = await Promise.all(tyeProcesses.map(process => this.getPortForProcess(process.pid)));
+        
+        return await Promise.all(tyeProcesses.map(process => this.getTyeProcessWithPorts(process.pid)));
+    }
 
-        function hasValidPort(process: ProspectiveTyeProcess): process is TyeProcess {
-            return !!process.dashboardPort;
-        }
+    private async getTyeProcessesFromPorts(processes: TyeProcessWithPorts[]): Promise<TyeProcess[]> {
+        const allProcesses = await Promise.all(processes.map(process => this.getTyeProcessFromPorts(process)));
 
-        const tyeProcessesWithValidPorts =
-            tyeProcessesWithPorts
-                .filter(hasValidPort);
+        function isValidProcess(process: TyeProcess | undefined): process is TyeProcess {
+            return !!process;
+        }
 
-        return tyeProcessesWithValidPorts;
+        return allProcesses.filter(isValidProcess);
     }
 
-    private async getPortForProcess(pid: number): Promise<ProspectiveTyeProcess> {
+    private async getTyeProcessWithPorts(pid: number): Promise<TyeProcessWithPorts> {
         const ports = await this.portProvider.getPortsForProcess(pid);
 
-        return { pid, dashboardPort: ports[0] }
+        return { pid, ports };
     }
-}
 
+    private async getTyeProcessFromPorts(process: TyeProcessWithPorts): Promise<TyeProcess | undefined> {
```

Comment:
`determineDashboardPort` ?

The `getTyeProcessWithPorts` vs `getTyeProcessFromPorts` nomenclature is a little confusing...

Reply:
Yes, the difference was a little subtle.  Went with your suggestion.